src/app/components/detailed-page/components/detailed-page-roles/detailed-page-roles.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, Input } from '@angular/core';
+import { GroupedValues } from 'src/app/pipes/group-array-by/group-array-by.pipe';
+import { KnownForAsCredits } from 'src/app/types/person';
+import { FindFunc } from 'src/app/pipes/find/find.pipe';
+import { FilterFunc } from 'src/app/pipes/filter/filter.pipe';
+import { SortFunc } from 'src/app/pipes/sort/sort.pipe';
+@Component({
+  selector: 'mdb-detailed-page-roles',
+  templateUrl: './detailed-page-roles.component.html',
+  styleUrls: ['./detailed-page-roles.component.scss'],
+})
+export class DetailedPageRolesComponent {
+  @Input() public title?: string;
+  @Input() public rolesList: KnownForAsCredits[] = [];
+  @Input() public mainKnownFor = '';
+
+  public mainKnownForFindFunc(
+    mainKnownFor: string
+  ): FindFunc<GroupedValues<KnownForAsCredits>> {
+    return (item: GroupedValues<KnownForAsCredits>) =>
+      item.key === mainKnownFor;
+  }
+
+  public notMainKnownForFilterFunc(
+    mainKnownFor: string
+  ): FilterFunc<GroupedValues<KnownForAsCredits>> {
+    return (item: GroupedValues<KnownForAsCredits>) =>
+      item.key !== mainKnownFor;
+  }
+
+  public groupByAmountSortFunc: SortFunc<GroupedValues<KnownForAsCredits>> = (
+    a,
+    b
+  ) => {
+    return b.value.length - a.value.length;
+  };
+
+  public rolesByDateSortFunc: SortFunc<KnownForAsCredits> = (a, b) => {
+    const aDate = new Date(a.releaseDate);
```

Comment:
If you use Date object in mapping from Dto, then you can avoid this code here

Reply:
As mentioned above, Date isn't valid in case of empty strings, decided to switch thish field to string. Maybe there is better solution for that issue?

Reply:
The releaseDate was an optional property, do I right remember? Just make this in mapping function
'''
  releaseDate: dto.release_date ?? new Date(dto.release_date)
'''

Reply:
Updated, 
`'' || new Date('') => Invalid Date {}`,
 therefore used 
`'' ? new Date('') : undefined => undefined`